client/state/notices/account-recovery/index.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { errorNotice } from 'state/notices/actions';
+import { dispatchError } from '../utils';
+
+import {
+	ACCOUNT_RECOVERY_SETTINGS_FETCH_FAILED,
+	ACCOUNT_RECOVERY_SETTINGS_UPDATE_FAILED,
+	ACCOUNT_RECOVERY_SETTINGS_DELETE_FAILED,
+} from 'state/action-types';
+
+const getUpdateErrorMessage = ( target ) => {
+	switch ( target ) {
+		case 'phone':
+			return translate( 'An error occurred while updating your account recovery phone number.' );
+		case 'email':
+			return translate( 'An error occurred while updating your account recovery email.' );
+		default:
+			return translate( 'An error occurred while updating your account recovery options.' );
+	}
+};
+
+const getDeleteErrorMessage = ( target ) => {
+	switch ( target ) {
+		case 'phone':
+			return translate( 'An error occurred while deleting your account recovery phone number.' );
+		case 'email':
+			return translate( 'An error occurred while deleting your account recovery email.' );
+		default:
+			return translate( 'An error occurred while deleting your account recovery options.' );
+	}
+};
+
+const handlers = {
+	[ ACCOUNT_RECOVERY_SETTINGS_FETCH_FAILED ]: dispatchError(
+		translate( 'An error occurred while fetching for your account recovery settings.' )
+	),
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_FAILED ]: ( dispatch, { target } ) => dispatch( errorNotice( getUpdateErrorMessage( target ) ) ),
```

Comment:
It is because `dispatchError` is a utility function very suitable for the case of `ACCOUNT_RECOVERY_SETTINGS_FETCH_FAILED`, which no extra action properties need to be extracted. However, in case of `ACCOUNT_RECOVERY_SETTINGS_UPDATE_FAILED` and `ACCOUNT_RECOVERY_SETTINGS_DELETE_FAILED`, it would become:
```
( dispatch, { target } ) => dispatchError( getUpdateErrorMessage( target ) )( dispatch )
```
which I think that it is not clearer than just using `errorNotice`.

Lastly, what happens in ef011a61fb9ba51e1af71aa062d22e4b5f36dcc5 is to address [Jordan's comment](https://github.com/Automattic/wp-calypso/pull/9173#discussion_r87335937). I took it since I think that would make the function simpler. 

Does that answer your question? :)